packages/@ember/object/type-tests/mixin/index.test.ts

```diff
@@ -0,0 +1,9 @@
+import { expectTypeOf } from 'expect-type';
+
+import Mixin from '@ember/object/mixin';
+
+let newMixin = Mixin.create({
+  foo: 'bar',
+});
+
+expectTypeOf(newMixin).toMatchTypeOf<Mixin>();
```

Comment:
This is definitely an interesting case we'll need to think about in terms of the Classic support package when we do that. Supporting `.extend()` is one thing, but technically supporting `Mixin.create()` goes with it… which seems 😬 🤔 (This is totally fine and good for *this* PR, of course.)